chunk/chunk_store.go

```diff
@@ -161,6 +161,8 @@ func (c *Store) updateIndex(ctx context.Context, userID string, chunks []Chunk)
 // calculateDynamoWrites creates a set of batched WriteRequests to dynamo for all
 // the chunks it is given.
 func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch, error) {
+	seenIndexEntries := map[string]bool{}
```

Comment:
Another way of doing this is `map[string]struct{}{}` i.e. a map from strings to empty structs. No need to change this though, `bool` is also fine.